src/delegate.rs

```diff
@@ -98,6 +98,12 @@ pub trait LanguageServerCore {
 
     #[rpc(name = "textDocument/documentHighlight", raw_params)]
     fn document_highlight(&self, params: Params) -> BoxFuture<Option<Vec<DocumentHighlight>>>;
+
+    #[rpc(name = "textDocument/signatureHelp", raw_params)]
+    fn signature_help(&self, params: Params) -> BoxFuture<Option<SignatureHelp>>;
+
+    #[rpc(name = "textDocument/implementation", raw_params)]
+    fn goto_implementation(&self, params: Params) -> BoxFuture<Option<GotoImplementationResponse>>;
```

Comment:
This method should be moved to immediately below `LanguageServerCore::goto_type_definition()`.